src/Actions/AuthActions.js

```diff
@@ -0,0 +1,50 @@
+import axios from "axios";
+
+const BASE_URL = "https://api.openconceptlab.org/";
+
+// Action type
+const AUTHENTICATED = "loggedin_user";
+const AUTHENTICATION_FAILED = "login_failed";
+
+// Action creator
+const Login = response => {
+  return {
+    type: AUTHENTICATED,
+    payload: response.data
+  };
+};
+
+const LoginFailed = errorMessage => {
+  return {
+    type: AUTHENTICATION_FAILED,
+    payload: { errorMessage }
+  };
+};
+
+// login action
+const LoginAction = ({ username, token }) => {
+  return async dispatch => {
+    try {
+      const url = `${BASE_URL}/users/${username}/`,
+        headers = { Authorization: `Token ${token}` };
+      // dispatch login sent
+      const response = await axios.post(url, null, { headers });
+      dispatch(Login(response));
+      localStorage.setItem("token", `Token ${token}`);
+    } catch (error) {
+      if (error.response) {
+        dispatch(LoginFailed(error.response.data.detail));
+      } else if (error.request) {
+        dispatch(LoginFailed("Request can't be made"));
+      }
+    }
+  };
+};
+
+export {
+  AUTHENTICATED,
+  AUTHENTICATION_FAILED,
+  Login,
+  LoginFailed,
+  LoginAction
+};
```

Comment:
Looks good, but the **action types**, **action creators** and **async action creators could be been in different files**, you can imagine if you had _10_ action types and more async action creators.

Reply:
Thanks @mwaz @emasys I am going to do that

Reply:
@emasys I have resolved your feedback, thanks